dashboard/modules/state/state_head.py

```diff
@@ -70,6 +70,21 @@ async def _update_raylet_stubs(self, change: Change):
                 int(node_info["nodeManagerPort"]),
             )
 
+    async def _update_agent_stubs(self, change: Change):
+        """Callback that's called when a new agent is added to Datasource."""
```

Comment:
This looks like nearly the exact same logic you added in the other PR, is it possible to deduplicate them?

Reply:
Let me try, but I think it might not be worth it (could be over-abstraction to deduplicate). We won't have anything like this other than these two, so it might be okay? I